code/datums/gamemodes/infestation.dm

```diff
@@ -6,3 +6,10 @@
 		return
 	var/datum/job/scaled_job = SSjob.GetJobType(/datum/job/terragov/squad/smartgunner)
 	scaled_job.job_points_needed  = 20 //For every 10 marine late joins, 1 extra SG
+
+/datum/game_mode/infestation/scale_roles()
```

Comment:
you are duplicating a proc that exist already. Add code the proc that's just above